jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -421,13 +422,38 @@ private static void configureContainerizer(
         .setToolName(projectProperties.getToolName())
         .setEventHandlers(projectProperties.getEventHandlers())
         .setAllowInsecureRegistries(rawConfiguration.getAllowInsecureRegistries())
-        .setBaseImageLayersCache(Containerizer.DEFAULT_BASE_CACHE_DIRECTORY)
-        .setApplicationLayersCache(projectProperties.getCacheDirectory());
+        .setBaseImageLayersCache(
+            getCheckedCacheDirectory(
+                PropertyNames.BASE_IMAGE_LAYERS_CACHE, Containerizer.DEFAULT_BASE_CACHE_DIRECTORY))
+        .setApplicationLayersCache(
+            getCheckedCacheDirectory(
+                PropertyNames.APPLICATION_IMAGE_LAYERS_CACHE,
+                projectProperties.getDefaultCacheDirectory()));
 
     rawConfiguration.getToTags().forEach(containerizer::withAdditionalTag);
 
     if (rawConfiguration.getUseOnlyProjectCache()) {
-      containerizer.setBaseImageLayersCache(projectProperties.getCacheDirectory());
+      containerizer.setBaseImageLayersCache(
+          getCheckedCacheDirectory(
+              PropertyNames.BASE_IMAGE_LAYERS_CACHE, projectProperties.getDefaultCacheDirectory()));
+    }
+  }
+
+  /**
+   * Returns the value of a cache directory system property if it is set, otherwise returns {@code
+   * defaultPath}.
+   *
+   * @param property the name of the system property to check
+   * @param defaultPath the path to return if the system property isn't set
+   * @return the value of a cache directory system property if it is set, otherwise returns {@code
+   *     defaultPath}
+   */
+  private static Path getCheckedCacheDirectory(String property, Path defaultPath) {
+    if (System.getProperty(property) != null) {
+      Path path = Paths.get(System.getProperty(property));
+      return path.isAbsolute() ? path : Paths.get(System.getProperty("user.dir")).resolve(path);
```

Comment:
Hmm, why is this resolving relative to the user dir (how is it different from just using the path directly?)?

Reply:
Because I don't know how paths work, I guess.